parquet-variant/src/encoder/variant_to_json.rs

```diff
@@ -0,0 +1,993 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Module for converting Variant data to JSON format
+
+use arrow_schema::ArrowError;
+use base64::{engine::general_purpose, Engine as _};
+use serde_json::Value;
+use std::io::Write;
+
+use crate::variant::{Variant, VariantList, VariantObject};
+
+/// Converts a Variant to JSON and writes it to the provided buffer
+///
+/// # Arguments
+///
+/// * `json_buffer` - Writer to output JSON to
+/// * `variant` - The Variant value to convert
+///
+/// # Returns
+///
+/// * `Ok(())` if successful
+/// * `Err` with error details if conversion fails
+///
+/// # Example
+///
+/// ```rust
+/// use parquet_variant::{Variant, variant_to_json};
+/// use arrow_schema::ArrowError;
+///
+/// fn example() -> Result<(), ArrowError> {
+///     let variant = Variant::Int8(42);
+///     let mut buffer = Vec::new();
+///     variant_to_json(&mut buffer, &variant)?;
+///     assert_eq!(String::from_utf8(buffer).unwrap(), "42");
+///     Ok(())
+/// }
+/// example().unwrap();
```

Comment:
I think you can make this significantly shorter like this -- the `#` in the front hides the line when rendered as docs and the test driver will make the stub program necessary

```suggestion
/// # use parquet_variant::{Variant, variant_to_json};
/// # use arrow_schema::ArrowError;
/// let variant = Variant::Int8(42);
/// let mut buffer = Vec::new();
/// variant_to_json(&mut buffer, &variant)?;
```

Also I think it would be awesome now to show an example of a VariantObject here (which is the most likely type that people will encode)

We could use the new builder that @PinkCrow007  added in https://github.com/apache/arrow-rs/pull/7653

Here is an example:

https://github.com/apache/arrow-rs/blob/fe65b8d9377cb9734cc2be8e4517492fa7eba867/parquet-variant/src/builder.rs#L90-L124